src/module.c

```diff
@@ -13489,6 +13492,111 @@ int RM_GetDbIdFromDefragCtx(RedisModuleDefragCtx *ctx) {
     return ctx->dbid;
 }
 
+/* --------------------------------------------------------------------------
+ * ## Key API for Set type
+ *
+ * See also RM_ValueLength(), which returns the length of a set.
+ * -------------------------------------------------------------------------- */
+
+/* Add new elements into a set.
+ *
+ * On success the function returns REDISMODULE_OK. On the following errors
+ * REDISMODULE_ERR is returned:
+ *
+ * * The key was not opened for writing.
+ * * The key is of the wrong type.
+ *
+ * In order to know the number of elements were added, the additional argument
+ * 'added' must be passed, that populates the integer by reference
+ * setting it to the number of elements added depending on the outcome of the operation.
+ * The 'added' argument can be NULL if the caller is not interested
+ * to know if the number of elements were really added.
+ *
+ * Empty keys will be created with set key type and continue. */
+int RM_SetAdd(RedisModuleKey *key, RedisModuleString **eles, size_t numeles, size_t *added) {
```

Comment:
i wonder if we should use a similar interface as RM_HashSet, e.g. one that's able to take c-strings too?
i'm not particularly fond of that API, but we should also take consistency into considerations.

Reply:
Good idea. Let me figure out how to realize it like RM_HashSet way.

Reply:
Now, I changed using fmt ("vslc") method.

Reply:
@zh1029 Not sure I'm right, we shouldn't change the usage of `moduleCreateArgvFromUserFormat` which is just used for module APIs related to command argv(cmdname + argv).

Reply:
Yes. I expended moduleCreateArgvFromUserFormat usage since reusing the fmt("vslc") to avoid code duplication.

Reply:
I can write new function if you don't think it is good.

Reply:
But IMHO since moduleCreateArgvFromUserFormat is internal function, from avoid code duplication perspective, expend the usage of moduleCreateArgvFromUserFormat may be not a problem.

Reply:
@zh1029 Agree with you, this change is better than duplication.

Reply:
I changed moduleCreateArgvFromUserFormat to moduleCreateArgvFromUserFormatAux.
then `robj **moduleCreateArgvFromUserFormat(const char *cmdname, const char *fmt, int *argcp, int *flags, va_list ap)` is used by RM_Call(), RM_Replicate()... which return flags
while `robj **moduleCreateSetArgvFromUserFormat(const char *fmt, int *argcp, va_list ap)` is used by RM_SetAdd(), RM_SetRem()... which doesn't return flags.

Both moduleCreateArgvFromUserFormat() and moduleCreateSetArgvFromUserFormat() invokes moduleCreateArgvFromUserFormatAux()

Maybe it is better now.

Reply:
@zh1029 Make sense to me.